components/identity-extensions/backend-oauth-authenticator/src/main/java/org/wso2/carbon/identity/authenticator/backend/oauth/validator/OAuthValidatorFactory.java

```diff
@@ -0,0 +1,46 @@
+/*
+*  Copyright (c) 2015 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing,
+*  software distributed under the License is distributed on an
+*  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+*  KIND, either express or implied.  See the License for the
+*  specific language governing permissions and limitations
+*  under the License.
+*/
+package org.wso2.carbon.identity.authenticator.backend.oauth.validator;
+
+import org.wso2.carbon.identity.authenticator.backend.oauth.AuthenticatorException;
+import org.wso2.carbon.identity.authenticator.backend.oauth.OauthAuthenticatorConstants;
+import org.wso2.carbon.identity.authenticator.backend.oauth.validator.impl.ExternalOAuthValidator;
+import org.wso2.carbon.identity.authenticator.backend.oauth.validator.impl.LocalOAuthValidator;
+
+/**
+ * The class validate the configurations and provide the most suitable implementation according to the configuration.
+ * Factory class for OAuthValidator.
+ */
+public class OAuthValidatorFactory {
+
+    /**
+     * The method check the configuration and provide the appropriate implementation for OAuth2TokenValidator
+     * @return OAuth2TokenValidator
+     */
+    public static OAuth2TokenValidator getValidator(boolean isRemote, String hostURL) throws IllegalArgumentException {
+        if (isRemote) {
+            if (!(hostURL == null || hostURL.trim().isEmpty())) {
+                hostURL = hostURL + OauthAuthenticatorConstants.OAUTH_ENDPOINT_POSTFIX;
+                return new ExternalOAuthValidator(hostURL);
+            } else {
+                throw new IllegalArgumentException("Remote server name and ip both can't be empty");
```

Comment:
Programmatically what we are getting is a host url for the EMM's side. For the end user we have provided the configuration in two different ways to increase the usability and user experience, to configure as hostname if the servers are hosted with the domain name and to configure as server IP if the servers are hosted locally or with a static IP. Host URL will be null when the user kept both of the config parameters empty in the authenticators.xml